python/pyiceberg/expressions/visitors.py

```diff
@@ -754,6 +752,6 @@ def visit_bound_predicate(self, predicate: BoundPredicate[Any]) -> BooleanExpres
 
 
 def inclusive_projection(
-    expr: BooleanExpression, schema: Schema, spec: PartitionSpec, case_sensitive: bool = True
-) -> BooleanExpression:
-    return visit(expr, InclusiveProjection(schema, spec, case_sensitive))
+    schema: Schema, spec: PartitionSpec, case_sensitive: bool = True
+) -> Callable[[BooleanExpression], BooleanExpression]:
```

Comment:
I opted to return a callable so that this works like the other methods.

Reply:
I like it 👍🏻